File: content/82/a.cpp
#include <iostream>
#include <numeric>
#include <vector>

std::string f(std::string s_acc, std::string s_cur) {
   return s_acc + s_cur;
}

int main() {
   std::vector<std::string> a = {"May", "June"};
   auto s = std::accumulate(a.begin(), a.end(), std::string(), f);
   std::cout << s << std::endl;
}

File: content/82/a.go
package main
import "fmt"
type flip map[string]int

func f(m_acc flip, s_cur string, n_idx int) flip {
   m_acc[s_cur] = n_idx
   return m_acc
}

func main() {
   a := []string{"May", "June"}
   m := flip{}
   for n, s := range a {
      m = f(m, s, n)
   }
   fmt.Println(m)
}
